### PR TITLE
Allow pointer comparisons and KVO methods to avoid initiating formal accesses

### PR DESCRIPTION
**Pull request by rjmccall**

### CODE REVIEW

**Comment by rjmccall** [comment]
@swift-ci Please test.

**Comment by rjmccall** [comment]
@jrose-apple Please tell me this is terrible, and then tell me whether it's okay anyway. :)

**Comment by rjmccall** [comment]
@devincoughlin @atrick You are probably interested in this.  I think having no markers in Swift 4 is probably reasonable, but post-4 we probably should add explicit markers that suppress static checking, just to allow a hard formation rule that the markers are always present.

**Comment by rjmccall** [comment]
Well, landing, but totally open to post-commit review.

File: lib/AST/Identifier.cpp [hunk 1]
```diff
@@ -105,6 +105,30 @@ int DeclName::compare(DeclName other) const {
...
+          getBaseName() == baseName &&
+          equals(getArgumentNames(), argNames));
+}
+
+bool DeclName::isCompoundName(StringRef baseName,
+                              ArrayRef<StringRef> argNames) const {
+  return (isCompoundName() &&
+          getBaseName() == baseName &&
+          equals(getArgumentNames(), argNames));
+}
```
Comment (jrose-apple) [inline lib/AST/Identifier.cpp]: These bits ended up in the wrong commit, but I guess that's what merge commits are for.

**Review by jrose-apple — COMMENTED** [review]
"This is terrible and it's probably the best thing we can do at this point."

**Comment by rjmccall** [comment]
Ok, thanks!  I'll open a 4.0 PR.

**Pull request merged**